### DISCUSSION TITLE
Transparency when no windows are present (only single frame)

### DISCUSSION THREAD

**Discussion by RayZ0rr** [herbstluftwm/herbstluftwm#1426] [General]
<details>
<summary>herbstluftwm --version</summary>

```
herbstluftwm 0.9.3
Copyright (c) 2011-2021 Thorsten Wißmann
Released under the Simplified BSD License
xrandr support: on
xinerama support: on
```
</details>

You can control transparency with two different set of settings (for above mentioned version at least) : 

1)
```sh
hc set frame_transparent_width 0
hc set always_show_frame off
hc set frame_bg_transparent on
hc set smart_frame_surroundings on
```
![desktop](https://user-images.githubusercontent.com/96843787/151210435-5d95d5ec-4283-426c-987f-a807f9da2c9e.png)

**This will give true transparency when only a single empty frame is present**. This would've been fine but now for single windows in single frames `hc set frame_gap 10` will be ignored. If you want to keep that setting, then there's following setting available.
2)
```sh
hc set frame_bg_normal_color '#00000000'
hc set frame_bg_active_color '#00000000'
hc set frame_transparent_width 0
hc set frame_border_width 5
hc set always_show_frame off
hc set frame_bg_transparent off   # --->  this doesn't play nicely with frame_bg_?_color settings
hc set smart_frame_surroundings off
```
But this gives an empty frame like this (inside is blurred): 

![transparency_bug1](https://user-images.githubusercontent.com/96843787/151208295-e3780fc0-4df1-496b-a5d2-f5ccd4d00a8d.png)
Changing the above setting to : 
```sh
hc set frame_bg_normal_color '#00000000'
hc set frame_bg_active_color '#00000000'
hc set frame_transparent_width 0
hc set frame_border_width 5
hc set always_show_frame off
hc set frame_bg_transparent on   # <----- Switch on
hc set smart_frame_surroundings off
```
![transparency_bug2](https://user-images.githubusercontent.com/96843787/151208688-4c231bc3-1466-43a3-a032-8bac647d3f74.png)
The best we can get with `smart_frame_surroundings` off is : 
```sh
hc set frame_bg_normal_color '#00000000'
hc set frame_bg_active_color '#00000000'
hc set frame_transparent_width 0
hc set frame_border_width o   # <----- change to 0
hc set always_show_frame off
hc set frame_bg_transparent off   # <----- Switch off
hc set smart_frame_surroundings off
```
![transparency_bug3](https://user-images.githubusercontent.com/96843787/151208959-5021c4a2-0a48-43de-bbd3-ec0cd02a07cb.png)

The frame is blurred. I changed all sort of setting in picom but it still was blurred.

There's this relevant issue that I answered myself : https://github.com/herbstluftwm/herbstluftwm/issues/1378

**Comment by t-wissmann** [comment]
I suggest that you add 
```
blur-background-exclude = [
    "class_g = '_HERBST_FRAME'",
]
shadow-exclude = [
    "class_g = '_HERBST_FRAME'",
]
shadow-ignore-shaped = true;
```
to your picom conf. Then you can go for (as in the third screenshot).
```
hc set frame_bg_normal_color '#00000000'
hc set frame_bg_active_color '#00000000'
hc set frame_transparent_width 0
hc set frame_border_width 5
hc set always_show_frame off
hc set frame_bg_transparent on
hc set smart_frame_surroundings off
```
The grey area you see is the shadows. You can remove this shadow with the picom option:
```
clear-shadow = true;
```

**Reply by RayZ0rr** [reply]
Thanks for the tip. I put `"class_g = 'herbstluftwm'",` in all my excludes and finally gave up. What you suggested indeed does the trick. However, `clear-shadow = true;` doesn't seem to have any effect. I will keep it anyway.
